pkg/resources/oauth_integration_for_custom_clients.go

```diff
@@ -0,0 +1,666 @@
+				return diag.FromErr(err)
+			}
+
+			oauthRefreshTokenValidity, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "OAUTH_REFRESH_TOKEN_VALIDITY"
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			if err = handleExternalChangesToObjectInDescribe(d,
+				describeMapping{"oauth_allow_non_tls_redirect_uri", "oauth_allow_non_tls_redirect_uri", oauthAllowNonTlsRedirectUri.Value, oauthAllowNonTlsRedirectUri.Value, nil},
+				describeMapping{"oauth_enforce_pkce", "oauth_enforce_pkce", oauthEnforcePkce.Value, oauthEnforcePkce.Value, nil},
+				describeMapping{"oauth_use_secondary_roles", "oauth_use_secondary_roles", oauthUseSecondaryRoles.Value, oauthUseSecondaryRoles.Value, nil},
+				describeMapping{"oauth_issue_refresh_tokens", "oauth_issue_refresh_tokens", oauthIssueRefreshTokens.Value, oauthIssueRefreshTokens.Value, nil},
+				describeMapping{"oauth_refresh_token_validity", "oauth_refresh_token_validity", oauthRefreshTokenValidity.Value, oauthRefreshTokenValidity.Value, nil},
+			); err != nil {
+				return diag.FromErr(err)
+			}
+		}
+
+		if err = setStateToValuesFromConfig(d, oauthIntegrationForCustomClientsSchema, []string{
+			"enabled",
+			"oauth_allow_non_tls_redirect_uri",
+			"oauth_enforce_pkce",
+			"oauth_use_secondary_roles",
+			"oauth_issue_refresh_tokens",
+			"oauth_refresh_token_validity",
+		}); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err = d.Set(ShowOutputAttributeName, []map[string]any{schemas.SecurityIntegrationToSchema(integration)}); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err = d.Set(DescribeOutputAttributeName, []map[string]any{schemas.DescribeOauthIntegrationForCustomClientsToSchema(integrationProperties)}); err != nil {
+			return diag.FromErr(err)
+		}
+
+		return nil
+	}
+}
+
+func UpdateContextOauthIntegrationForCustomClients(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+	set, unset := sdk.NewOauthForCustomClientsIntegrationSetRequest(), sdk.NewOauthForCustomClientsIntegrationUnsetRequest()
+
+	if d.HasChange("oauth_redirect_uri") {
+		set.WithOauthRedirectUri(d.Get("oauth_redirect_uri").(string))
+	}
+
+	if d.HasChange("enabled") {
+		if v := d.Get("enabled").(string); v != BooleanDefault {
+			parsedBool, err := strconv.ParseBool(v)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			set.WithEnabled(parsedBool)
+		} else {
+			unset.WithEnabled(true)
+		}
+	}
+
+	if d.HasChange("oauth_allow_non_tls_redirect_uri") {
+		if v := d.Get("oauth_allow_non_tls_redirect_uri").(string); v != BooleanDefault {
+			parsedBool, err := strconv.ParseBool(v)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			set.WithOauthAllowNonTlsRedirectUri(parsedBool)
+		} else {
+			// No unset available for this field (setting Snowflake default)
```

Comment:
Pls mention SNOW-1515781. In other places as well.

Reply:
Done